modules/fluid_properties/src/fluidproperties/HeliumFluidProperties.C

```diff
@@ -30,6 +30,36 @@ HeliumFluidProperties::fluidName() const
   return "helium";
 }
 
+Real
+HeliumFluidProperties::e_from_p_rho(Real p, Real rho) const
+{
+  // Initial guess using ideal gas law
+  Real e = p / (_cp / _cv - 1.) / rho;
+  const Real v = 1. / rho;
+
+  Real p_from_props, dp_dv, dp_de;
+  const unsigned int max_its = 10;
+  unsigned int it = 0;
+
+  do
+  {
+    p_from_v_e(v, e, p_from_props, dp_dv, dp_de);
+    const Real & jacobian = dp_de;
+    const Real residual = p_from_props - p;
+
+    if (std::abs(residual) / p < 1e-12)
+      break;
+
+    const Real delta_e = -residual / jacobian;
+    e += delta_e;
+  } while (++it < max_its);
+
+  if (it >= max_its)
+    mooseWarning("The e_from_p_rho iteration failed to converge");
```

Comment:
until there is a consistent strategy for dealing with failing fluidprops, this is as good as we can do without derailing people s simulations when Newton goes overboard

Reply:
Ah, right. The other option is to return a NaN with `NaNInterface`. I'll leave the choice up to you.